lib/ansible/modules/cloud/vmware/vmware_guest_video.py

```diff
@@ -0,0 +1,325 @@
+                break
+        return video_card, video_facts
+
+    def get_video_card_spec(self, vm_obj):
+        """
+        Get device changes of virtual machine
+        Args:
+            vm_obj: Managed object of virtual machine
+        Returns: virtual device spec
+        """
+        video_card, video_card_facts = self.gather_video_card_facts(vm_obj)
+        self.video_card_facts = video_card_facts
+        if video_card is None:
+            self.module.fail_json(msg='Not get video card device of specified virtual machine.')
+        video_spec = vim.vm.device.VirtualDeviceSpec()
+        video_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
+        video_spec.device = video_card
+        auto_detect = False
+        enabled_3d = False
+
+        if self.params['gather_video_facts']:
+            return None
+        if self.params['use_auto_detect'] is not None:
+            if video_card_facts['auto_detect'] and self.params['use_auto_detect']:
+                auto_detect = True
+            elif not video_card_facts['auto_detect'] and self.params['use_auto_detect']:
+                video_spec.device.useAutoDetect = True
+                self.change_detected = True
+                auto_detect = True
+            elif video_card_facts['auto_detect'] and not self.params['use_auto_detect']:
+                video_spec.device.useAutoDetect = False
+                self.change_detected = True
+        else:
+            if video_card_facts['auto_detect']:
+                auto_detect = True
+        # useAutoDetect set to False then display number and video memory config can be changed
+        if not auto_detect:
+            if self.params['display_number'] is not None:
+                if self.params['display_number'] < 1:
+                    self.module.fail_json(msg="display_number attribute valid value: 1-10.")
+                if self.params['display_number'] != video_card_facts['display_number']:
+                    video_spec.device.numDisplays = self.params['display_number']
+                    self.change_detected = True
+
+            if self.params['video_memory_mb'] is not None:
+                if self.params['video_memory_mb'] < 2:
+                    self.module.fail_json(msg="video_memory_mb attribute valid value: ESXi 6.7(1.172-256 MB),"
+                                              "ESXi 6.5/6.0(1.17-128 MB).")
+                if self.params['video_memory_mb'] * 1024 != video_card_facts['video_memory']:
+                    video_spec.device.videoRamSizeInKB = self.params['video_memory_mb'] * 1024
+                    self.change_detected = True
+        else:
+            if self.params['display_number'] is not None or self.params['video_memory_mb'] is not None:
+                self.module.fail_json(msg="display_number and video_memory_mb can not be changed if use_auto_detect is true.")
+        # useAutoDetect value not control 3D config
+        if self.params['enable_3D'] is not None:
```

Comment:
`enable_3D` is a bool, so it will always be set, i think this can be a little simplified as well
```suggestion
        video_spec.device.enable3DSupport = self.params['enabled_3D']
        enabled_3d = self.params['enabled_3D']
        if self.params['enable_3D'] != video_card_facts['enable_3D_support']:
            self.change_detected = True
```

Reply:
"enable_3D" is also not a must-have attribute. If it's not specified, "self.params['enable_3D']" will be None.

Reply:
You are absolutely right :). Just did a quick test... sorry for my comment.

Reply:
Thank you for your quick reviewing and checking:), I'll test a little bit more on vCenter and try to find a better way to handle that "float 1.172" value issue above. Thanks a lot.